Kernel/InterruptHandler.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <AK/Types.h>
+
+class InterruptEnabler;
+class InterruptHandler {
+    friend class InterruptEnabler;
+public:
+    virtual ~InterruptHandler();
+    virtual void handle_irq() = 0;
+
+    u8 irq_number() const { return m_irq_number; }
+    bool is_enabled();
+
+protected:
+    void enable_irq();
+    void change_irq_number(u8 irq_number);
+    void disable_irq();
+    explicit InterruptHandler(u8 irq);
+
+private:
+    bool m_enabled { false };
+    u8 m_irq_number { 0 };
+};
+
+class InterruptEnabler {
```

Comment:
This name clashes too much with `InterruptDisabler` and that's not good since they are not the inverse of each other.

Do we really need this class in the first place? Temporarily enabling interrupts doesn't seem like a common pattern. I only see a single use in the next commit.

Reply:
It will become more common later when we'll add hardware support.
I agree the name isn't the best, what about `InterruptHandlingEnabler`?

Reply:
How will that pattern look like? The way you are using it to temporarily enable interrupts around a wait queue seems like it could be rolled into a helper function (that takes care of enable + wait + disable) instead.

Reply:
In the future I'm not sure I'll use the wait queue to schedule writings and readings, since that queue isn't suitable (or optimized) for advanced hardware protocols.
I don't think there is a problem with creating such class for now, even though it looks kinda 'useless' right now. Anyway, that's your call.

Reply:
I found that I can use such object also in the SB16 driver, in `SB16::wait_for_irq()`.

Reply:
Ok, let's just get away from the clash with `InterruptDisabler`. How about `DeviceInterruptEnabler`? To clarify that it's only for a specific device.

Reply:
That was solved, thanks kling :)